components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -626,6 +626,12 @@ public Group createGroup(Group group) throws CharonException, DuplicateResourceE
                     log.info("Group: " + group.getDisplayName() + " is created through SCIM.");
                 }
             } catch (UserStoreException e) {
+                try {
+                    SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
+                    scimGroupHandler.deleteGroupAttributes(group.getDisplayName());
+                } catch (UserStoreException | IdentitySCIMException ex) {
+                    throw new CharonException("Error occurred while doing rollback operation of the SCIM table entry for role: " + group.getDisplayName(), e);
```

Comment:
Better solution would be to throw the original exception and log the exception occurred while deleting. Client would be interested in knowing the original exception not the exception occurred during deletion.